api/v1beta2/openstackcontrolplane_webhook.go

```diff
@@ -0,0 +1,325 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Generated by:
+//
+// operator-sdk create webhook --group osp-director --version v1beta2 --defaulting --programmatic-validation --conversion --kind OpenStackControlPlane --force
+//
+
+package v1beta2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openstack-k8s-operators/osp-director-operator/api/shared"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	ospdirectorv1beta1 "github.com/openstack-k8s-operators/osp-director-operator/api/v1beta1"
+)
+
+var openstackControlPlaneDefaults shared.OpenStackControlPlaneDefaults
+
+// log is for logging in this package.
+var controlplanelog = logf.Log.WithName("controlplane-resource")
+
+// SetupWebhookWithManager - register this webhook with the controller manager
+func (r *OpenStackControlPlane) SetupWebhookWithManager(mgr ctrl.Manager, defaults shared.OpenStackControlPlaneDefaults) error {
+
+	openstackControlPlaneDefaults = defaults
+
+	if webhookClient == nil {
+		webhookClient = mgr.GetClient()
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-osp-director-openstack-org-v1beta2-openstackcontrolplane,mutating=true,failurePolicy=fail,sideEffects=None,groups=osp-director.openstack.org,resources=openstackcontrolplanes,verbs=create;update,versions=v1beta1;v1beta2,name=mopenstackcontrolplane.kb.io,admissionReviewVersions=v1
```

Comment:
Similar question regarding `name` here in relation to verions

Reply:
Ah, but I did see that you had qualified the names of the old version's webhooks, so I guess leaving these as-is won't cause the webhooks to overwrite each other